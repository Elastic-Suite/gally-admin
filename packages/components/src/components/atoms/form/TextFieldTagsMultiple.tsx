import React, { ReactNode, useEffect, useState } from 'react'
import { styled } from '@mui/system'
import { FormHelperText, InputLabel } from '@mui/material'
import IonIcon from '../IonIcon/IonIcon'
import InfoTooltip from './InfoTooltip'
import FormControl from './FormControl'
import TextFieldTags from './TextFieldTags'
import Button from '../buttons/Button'

import {
  IOptionsTags,
  ITransformedLimitations,
} from '@elastic-suite/gally-admin-shared'
import DropDown from './DropDown'

const CustomMultipleTextFieldsTags = styled('div')(({ theme }) => ({
  display: 'flex',
  gap: theme.spacing(2.5),
  flexDirection: 'column',
}))

const CustomCloseTagsByOperator = styled('div')(({ theme }) => ({
  position: 'absolute',
  right: '-5px',
  top: '-5px',
  borderRadius: '50%',
  background: theme.palette.colors.neutral[600],
  color: 'white',
  display: 'flex',
  padding: theme.spacing(0.5),
  zIndex: '9',
  cursor: 'pointer',
}))

const CustomSelectOperator = styled('div')(({ theme }) => ({
  display: 'flex',
  gap: theme.spacing(1),
  alignItems: 'center',
  flexDirection: 'row',
}))

export interface ITextFIeldTagsForm {
  disabled?: boolean
  error?: boolean
  fullWidth?: boolean
  infoTooltip?: string
  helperText?: ReactNode
  helperIcon?: string
  label?: string
  margin?: 'none' | 'dense' | 'normal'
  required?: boolean
  options: IOptionsTags[]
}

export interface ITextFieldTag extends ITextFIeldTagsForm {
  value: ITransformedLimitations
  onChange: (operator: string | string[], data?: string[] | string) => void
}

function TextFieldTagsMultiple(props: ITextFieldTag): JSX.Element {
  const {
    value,
    onChange,
    disabled,
    required,
    error,
    fullWidth,
    helperText,
    helperIcon,
    label,
    margin,
    infoTooltip,
    options,
  } = props

  const optionsListAvailable: (IOptionsTags & { disabled?: boolean })[] =
    options.map((item) => {
      if (value[item.value]) {
        return { ...item, disabled: true }
      }
      return item
    })

  const [optionDefault, setOptionDefault] = useState<undefined | IOptionsTags>()

  useEffect(() => {
    setOptionDefault(
      optionsListAvailable.find(
        (item) => optionDefault?.value === item.value && !item?.disabled
      ) ?? optionsListAvailable.find((item) => !item?.disabled)
    )
  }, [optionsListAvailable, optionDefault])

  return (
    <FormControl error={error} fullWidth={fullWidth} margin={margin}>
      {Boolean(label || infoTooltip) && (
        <div style={{ marginBottom: '24px' }}>
          <InputLabel shrink required={required}>
            {label}
            {infoTooltip ? <InfoTooltip title={infoTooltip} /> : null}
          </InputLabel>
        </div>
      )}
      <CustomMultipleTextFieldsTags>
        {disabled ? (
          <TextFieldTags disabledValue="disabled" disabled />
        ) : (
          Object.entries(value).map((item) => {
            const option = options.find((it) => it.value === item[0])
            const newOptionsList = options.map((it) => {
              if (value[it.value] && item[0] !== it.value) {
                return { ...it, disabled: true }
              }
              return it
            })
            return (
              <div key={item[0]}>
                <DropDown
                  onChange={(newOption): void =>
                    onChange(item[0], newOption as string)
                  }
                  value={option?.value}
                  options={newOptionsList}
                  sx={{ marginBottom: 1 }}
                />
                <div style={{ position: 'relative' }}>
                  <CustomCloseTagsByOperator
                    onClick={(): void => onChange(item[0])}
                  >
                    <IonIcon
                      name="close"
                      style={{ fontSize: 14, padding: '0px' }}
                    />
                  </CustomCloseTagsByOperator>
                  <TextFieldTags
                    onChange={(a): void => onChange(item[0], a)}
                    value={item[1]}
                    placeholder={option?.label}
                  />
                </div>
              </div>
            )
          })
        )}
        {optionDefault && !disabled ? <CustomSelectOperator>
            <DropDown
              onChange={(newOption): void =>
                setOptionDefault(
                  optionsListAvailable.find((item) => item?.value === newOption)
                )
              }
              value={optionDefault?.value}
              options={optionsListAvailable}
            />
            <Button
              size="medium"
              onClick={(): void => onChange([optionDefault.value])}
            >
              Add
            </Button>
          </CustomSelectOperator> : null}
      </CustomMultipleTextFieldsTags>
      {Boolean(helperText) && (
        <FormHelperText error={error}>
          {Boolean(helperIcon) && (
            <IonIcon
              name={helperIcon as string}
              style={{ fontSize: 18, marginRight: 2 }}
            />
          )}
          {helperText}
        </FormHelperText>
      )}
    </FormControl>
  )
}

export default TextFieldTagsMultiple
