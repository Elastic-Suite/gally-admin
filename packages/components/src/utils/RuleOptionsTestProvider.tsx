import React, { ReactNode } from 'react'
import {
  IRuleEngineOperators,
  RuleAttributeType,
  getIri,
} from '@elastic-suite/gally-admin-shared'

import ruleEngineOperator from '../../public/mocks/rule_engine_operators.json'
import sourceFields from '../../public/mocks/source_fields.json'
import sourceFieldLabels from '../../public/mocks/source_field_labels.json'

import RuleOptionsProvider from '../components/stateful-providers/RuleOptionsProvider/RuleOptionsProvider'

interface IProps {
  children: ReactNode
}

const sourceFieldLabelsMap = new Map(
  sourceFieldLabels['hydra:member'].map((label) => [
    label.sourceField,
    label.label,
  ])
)

const fields = sourceFields['hydra:member'].map((field) => {
  const label = sourceFieldLabelsMap.get(getIri('source_fields', field.id))
  return {
    id: field.id,
    code: field.code,
    label: (label || field.defaultLabel) ?? '',
    type: field.type as RuleAttributeType,
  }
})

function RuleOptionsTestProvider(props: IProps): JSX.Element {
  const { children } = props
  return (
    <RuleOptionsProvider
      catalogId={-1}
      localizedCatalogId={-1}
      fields={fields}
      ruleOperators={ruleEngineOperator as IRuleEngineOperators}
    >
      {children}
    </RuleOptionsProvider>
  )
}

export default RuleOptionsTestProvider
