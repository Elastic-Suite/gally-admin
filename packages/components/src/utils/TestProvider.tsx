import React, { ReactNode } from 'react'

import catalogs from '../../public/mocks/catalog.json'
import metadata from '../../public/mocks/metadata.json'

import { catalogContext, metadataContext } from '../contexts'
import { useCatalogs } from '../hooks'

import OptionsProvider from '../components/stateful-providers/OptionsProvider/OptionsProvider'

interface IProps {
  children: ReactNode
}

function TestProvider(props: IProps): JSX.Element {
  const { children } = props
  const contextValueCatalog = useCatalogs(catalogs['hydra:member'])
  const contextValueMetadata = metadata['hydra:member']

  return (
    <OptionsProvider>
      <catalogContext.Provider value={contextValueCatalog}>
        <metadataContext.Provider value={contextValueMetadata}>
          {children}
        </metadataContext.Provider>
      </catalogContext.Provider>
    </OptionsProvider>
  )
}

export default TestProvider
