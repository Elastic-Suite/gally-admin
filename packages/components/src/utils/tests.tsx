import React, { ReactNode } from 'react'
import { RenderOptions, render, renderHook } from '@testing-library/react'
import { PreloadedState } from '@reduxjs/toolkit'
import { Bundle, api } from '@elastic-suite/gally-admin-shared'

import { AppStore, RootState, setupStore } from '../store'

import AppProvider from '../components/stateful-providers/AppProvider/AppProvider'

import TestProvider from './TestProvider'

export interface IExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: PreloadedState<RootState>
  store?: AppStore
}

export interface IRenderOutput {
  store: AppStore
}

const defaultState = {
  data: {
    api,
    bundles: [Bundle.VIRTUAL_CATEGORY],
    configurations: {
      'base_url/media': 'https://localhost/media/catalog/product/',
    },
    metadata: [
      {
        '@id': '/metadata/3',
        '@type': 'Metadata',
        id: 3,
        entity: 'product',
        sourceFields: ['/source_fields/217', '/source_fields/253'],
      },
      {
        '@id': '/metadata/4',
        '@type': 'Metadata',
        id: 4,
        entity: 'category',
        sourceFields: ['/source_fields/254', '/source_fields/286'],
      },
    ],
  },
  user: {
    requestedPath: '',
  },
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderWithProviders(
  ui: React.ReactElement,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/explicit-function-return-type
export function renderHookWithProviders<R>(
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  hook: (props: any) => R,
  {
    preloadedState = defaultState,
    store,
    ...renderOptions
  }: IExtendedRenderOptions = {}
) {
  store = store ?? setupStore(preloadedState)
  function Wrapper({ children }: { children: ReactNode }): JSX.Element {
    return (
      <AppProvider store={store}>
        <TestProvider>{children}</TestProvider>
      </AppProvider>
    )
  }
  return { store, ...renderHook(hook, { wrapper: Wrapper, ...renderOptions }) }
}
