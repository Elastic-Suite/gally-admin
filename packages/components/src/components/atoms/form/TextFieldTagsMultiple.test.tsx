import React from 'react'

import { renderWithProviders } from '../../../utils/tests'

import TextFieldTagsMultiple from './TextFieldTagsMultiple'
import { screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import {
  IOptionsTags,
  ISearchLimitations,
} from '@elastic-suite/gally-admin-shared'

const textOperatorOptions: IOptionsTags[] = [
  {
    id: 'eq',
    value: 'eq',
    label: 'is',
  },
  {
    id: '%like',
    value: '%like',
    label: 'starts with',
  },
  {
    id: '%like%',
    value: '%like%',
    label: 'contains',
  },
  {
    id: 'like%',
    value: 'like%',
    label: 'ends with',
  },
]

const searchLimitations: ISearchLimitations[] = [
  {
    '@id': '/boost_search_limitations/1',
    '@type': 'BoostSearchLimitation',
    operator: 'eq',
    queryText: 'salut',
  },
  {
    '@id': '/boost_search_limitations/1',
    '@type': 'BoostSearchLimitation',
    operator: 'eq',
    queryText: 'eq again',
  },
  {
    '@id': '/boost_search_limitations/1',
    '@type': 'BoostSearchLimitation',
    operator: '%like',
    queryText: 'TEST123',
  },
  {
    '@id': '/boost_search_limitations/1',
    '@type': 'BoostSearchLimitation',
    operator: '%like%',
    queryText: 'Hello',
  },
]

describe('TextFieldTagsMultiple', () => {
  it('sould match snapshot', () => {
    const { container } = renderWithProviders(
      <TextFieldTagsMultiple
        label="Label"
        options={textOperatorOptions}
        value={searchLimitations}
      />
    )
    expect(container).toMatchSnapshot()
  })

  it('should display error with showError prop when the field is required with missing value', () => {
    renderWithProviders(
      <TextFieldTagsMultiple
        options={textOperatorOptions}
        label="Label"
        required
        value={[
          {
            operator: 'eq',
            queryText: null,
          },
        ]}
        showError
      />
    )
    expect(screen.getByText('formError.valueMissing')).toBeInTheDocument()
  })

  it('should not display error without showError prop when the field is required with missing value', () => {
    renderWithProviders(
      <TextFieldTagsMultiple
        options={textOperatorOptions}
        label="Label"
        required
        value={[
          {
            operator: 'eq',
            queryText: null,
          },
        ]}
      />
    )
    expect(screen.queryByText('formError.erreur')).not.toBeInTheDocument()
  })

  it('should not display error with showError prop when the field is required and disabled with missing value', () => {
    renderWithProviders(
      <TextFieldTagsMultiple
        label="Label"
        required
        disabled
        options={textOperatorOptions}
        value={[
          {
            operator: 'eq',
            queryText: null,
          },
        ]}
        showError
      />
    )
    expect(screen.queryByText('formError.valueMissing')).not.toBeInTheDocument()
  })
})
