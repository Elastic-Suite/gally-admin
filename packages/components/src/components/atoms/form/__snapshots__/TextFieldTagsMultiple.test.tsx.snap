// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextFieldTagsMultiple sould match snapshot 1`] = `
<div>
  <div
    class="textFieldTagsMultiple css-vaijrj"
  >
    <div
      style="margin-bottom: 4px;"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-colorPrimary css-1j7os2p-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
      >
        Label
      </label>
    </div>
    <div
      class="css-1yqt158"
    >
      <div
        style="display: flex; flex-direction: row; gap: 8px;"
      >
        <div
          style="padding: 16px; border: 1px solid #e2e6f3; border-radius: 8px; display: flex; flex-direction: row; gap: 8px;"
        >
          <div
            class="MuiFormControl-root css-1girsym-MuiFormControl-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1qqsdnr-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
              >
                <div
                  class="InputText__Wrapper css-17fp4jk"
                >
                  <div
                    class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-of7gsx-MuiInputBase-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-ngazp5-MuiInputBase-input"
                      id=":r0:"
                      role="combobox"
                      spellcheck="false"
                      type="text"
                      value="is"
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="form.open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="form.open"
                        type="button"
                      >
                        <ion-icon
                          name="chevron-down"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="position: relative;"
          >
            <div
              class="css-1r61l91"
            >
              <ion-icon
                name="close"
                style="font-size: 14px; padding: 0px;"
              />
            </div>
            <div
              class="css-vaijrj"
            >
              <div
                class="css-1mxyr7i"
              >
                <div
                  class="css-r1vopi"
                >
                  <div
                    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                    role="button"
                    style="opacity: 1;"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      salut
                    </span>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <ion-icon
                        name="close"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                    role="button"
                    style="opacity: 1;"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      eq again
                    </span>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <ion-icon
                        name="close"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="css-u4p24i"
                  >
                    <div
                      class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
                    >
                      <div
                        class="InputText__Wrapper css-17fp4jk"
                      >
                        <div
                          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-z54rnl-MuiInputBase-root"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                            placeholder="placeholder.default.textFieldTags"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <ion-icon
              name="trash-outline"
              style="font-size: 18px; color: rgb(66, 72, 128);"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        style="display: flex; flex-direction: row; gap: 8px;"
      >
        <div
          style="padding: 16px; border: 1px solid #e2e6f3; border-radius: 8px; display: flex; flex-direction: row; gap: 8px;"
        >
          <div
            class="MuiFormControl-root css-1girsym-MuiFormControl-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1qqsdnr-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
              >
                <div
                  class="InputText__Wrapper css-17fp4jk"
                >
                  <div
                    class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-of7gsx-MuiInputBase-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-ngazp5-MuiInputBase-input"
                      id=":r1:"
                      role="combobox"
                      spellcheck="false"
                      type="text"
                      value="starts with"
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="form.open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="form.open"
                        type="button"
                      >
                        <ion-icon
                          name="chevron-down"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="position: relative;"
          >
            <div
              class="css-1r61l91"
            >
              <ion-icon
                name="close"
                style="font-size: 14px; padding: 0px;"
              />
            </div>
            <div
              class="css-vaijrj"
            >
              <div
                class="css-1mxyr7i"
              >
                <div
                  class="css-r1vopi"
                >
                  <div
                    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                    role="button"
                    style="opacity: 1;"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      TEST123
                    </span>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <ion-icon
                        name="close"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="css-u4p24i"
                  >
                    <div
                      class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
                    >
                      <div
                        class="InputText__Wrapper css-17fp4jk"
                      >
                        <div
                          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-z54rnl-MuiInputBase-root"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                            placeholder="placeholder.default.textFieldTags"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <ion-icon
              name="trash-outline"
              style="font-size: 18px; color: rgb(66, 72, 128);"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        style="display: flex; flex-direction: row; gap: 8px;"
      >
        <div
          style="padding: 16px; border: 1px solid #e2e6f3; border-radius: 8px; display: flex; flex-direction: row; gap: 8px;"
        >
          <div
            class="MuiFormControl-root css-1girsym-MuiFormControl-root"
          >
            <div
              class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon css-1qqsdnr-MuiAutocomplete-root"
            >
              <div
                class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
              >
                <div
                  class="InputText__Wrapper css-17fp4jk"
                >
                  <div
                    class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-of7gsx-MuiInputBase-root"
                  >
                    <input
                      aria-autocomplete="list"
                      aria-expanded="false"
                      aria-invalid="false"
                      autocapitalize="none"
                      autocomplete="off"
                      class="MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-ngazp5-MuiInputBase-input"
                      id=":r2:"
                      role="combobox"
                      spellcheck="false"
                      type="text"
                      value="contains"
                    />
                    <div
                      class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                    >
                      <button
                        aria-label="form.open"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator css-qzbt6i-MuiButtonBase-root-MuiIconButton-root-MuiAutocomplete-popupIndicator"
                        tabindex="-1"
                        title="form.open"
                        type="button"
                      >
                        <ion-icon
                          name="chevron-down"
                        />
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            style="position: relative;"
          >
            <div
              class="css-1r61l91"
            >
              <ion-icon
                name="close"
                style="font-size: 14px; padding: 0px;"
              />
            </div>
            <div
              class="css-vaijrj"
            >
              <div
                class="css-1mxyr7i"
              >
                <div
                  class="css-r1vopi"
                >
                  <div
                    class="MuiButtonBase-root MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-deletable MuiChip-deletableColorDefault MuiChip-filledDefault css-69f6vo-MuiButtonBase-root-MuiChip-root"
                    role="button"
                    style="opacity: 1;"
                    tabindex="0"
                  >
                    <span
                      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                    >
                      Hello
                    </span>
                    <button
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiChip-deleteIcon MuiChip-deleteIconMedium MuiChip-deleteIconColorDefault MuiChip-deleteIconFilledColorDefault css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                      tabindex="0"
                      type="button"
                    >
                      <ion-icon
                        name="close"
                      />
                      <span
                        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <div
                    class="css-u4p24i"
                  >
                    <div
                      class="MuiFormControl-root css-hh3vi8-MuiFormControl-root"
                    >
                      <div
                        class="InputText__Wrapper css-17fp4jk"
                      >
                        <div
                          class="MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-z54rnl-MuiInputBase-root"
                        >
                          <input
                            aria-invalid="false"
                            class="MuiInputBase-input css-ngazp5-MuiInputBase-input"
                            placeholder="placeholder.default.textFieldTags"
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          style="display: flex; align-items: center;"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <ion-icon
              name="trash-outline"
              style="font-size: 18px; color: rgb(66, 72, 128);"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
      <div
        class="css-1khnmui"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-6rpdpi-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Add
          <span
            class="MuiButton-endIcon MuiButton-iconSizeMedium css-1gnd1fd-MuiButton-endIcon"
          >
            <ion-icon
              name="add"
              style="font-size: 24px;"
            />
          </span>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
